api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -547,6 +561,33 @@ public DomainValuePair apply(DataSetValues dataSetValue) {
     return ResponseEntity.ok(TO_CLIENT_DATA_SET.apply(dbDataSet));
   }
 
+  // If the dataSet has concept set of Domain Person, check if such a concept set exist in workspace
+  // if not create a dummy Concept set with name Demographics and Domain PERSON.
+  private void handleDemographicConceptSet(
+      DataSetRequest request, long workspaceId, Timestamp now) {
+    if (request.getConceptSetIds().size() > 0 && request.getConceptSetIds().contains(-1l)) {
+      request.getConceptSetIds().remove(-1l);
+      // -1 Represent Demographics concepts set, Check for GENDER OR RACE OR ETHNICITY DOMAIN
+      // concept
+      // and add it as a dummy concept under the Demographics concept set
+      List<Concept> genderRaceConcept = conceptDao.findGenderRaceEthnicityFromConcept();
+      if (genderRaceConcept.size() > 0) {
+        org.pmiops.workbench.db.model.ConceptSet demographicsCS =
+            new org.pmiops.workbench.db.model.ConceptSet();
+        demographicsCS.setName("Demographics");
+        demographicsCS.setDomain(CommonStorageEnums.domainToStorage(Domain.PERSON));
+        demographicsCS.setCreationTime(now);
+        demographicsCS.setLastModifiedTime(now);
+        demographicsCS.setWorkspaceId(workspaceId);
+        Set<Long> conceptSet = new HashSet<Long>();
```

Comment:
Nit: Since we're already using concept set to refer to the object type, can we name this something like: `conceptsInSet`?

Reply:
How about conceptIdSets

Reply:
Sure, that seems fine.